### DISCUSSION TITLE
Pressure Fields

### DISCUSSION THREAD

**Discussion by s-aucoin** [CliMA/Oceananigans.jl#3157] [General]
Hello,

I have a question about what the variables pHY′ and pNHS that model.pressures (for a NonhydrostaticModel) gives you mean physically. I ran a test simulation with uniform density, no forcing, no background fields, to a depth of 11 m to try to understand the pressure outputs. The non-hydrostatic variable, pNHS was zero everywhere, but pHY′ ranged between about 0 at the surface, and about 1.8 at the seabed. From the [pressure decomposition page](https://clima.github.io/OceananigansDocumentation/stable/numerical_implementation/pressure_decomposition/#mjx-eqn%3Aeq%3Apressure), I assumed that pHY′ must either be the total hydrostatic pressure, or the hydrostatic anomaly (all normalized by $\rho_0$), but in my test case, $\rho = \rho_0$, so $\rho_{\star} = 0$ and $\rho^{\prime} = 0$, implying that the pressure should simplify to $P = -gz$. This doesn't seem to be the case, however; this figure shows the pHY′ variable with depth compared to separately calculated hydrostatic pressure functions. What I notice in particular is that both the slope and the intercept seem to be different, so there must be something I am missing in the $\partial_z p$ equation. What are the units and the physical meaning of pHY′ and pNHS? ![Pressures](https://github.com/CliMA/Oceananigans.jl/assets/137843770/c28393ec-b2d9-449d-88e2-2fe4dbc21a91)

**Comment by glwagner** [comment]
`pHY'` is not the total hydrostatic pressure, but rather the "hydrostatic pressure anomaly". This screenshot is from the page you linked to:

<img width="1064" alt="image" src="https://github.com/CliMA/Oceananigans.jl/assets/15271942/4bdbf4c9-a6cb-49d5-9576-f54872f9bb7a">

So the "hydrostatic pressure anomaly" is defined as

$$ p_h = \int_z^0 b \text{ d}z $$

where $b$ is the buoyancy anomaly. This definition comes from the Boussinesq vertical momentum equation:

$$ \partial_t w + \cdots = - \partial_z p + b + \cdots $$

It's implemented here:

https://github.com/CliMA/Oceananigans.jl/blob/ecb559fe9b84a76db514f8d26638abaef582aec4/src/Models/NonhydrostaticModels/update_hydrostatic_pressure.jl#L16

You should only have a hydrostatic pressure anomaly if you have specified some `buoyancy`, which is the dynamical quantity corresponding to variations in density in a Boussinesq model. If the buoyancy is uniform but non-zero (say $b=b_0$) then the pressure will be $p = - b_0 z$. But a uniform buoyancy is dynamically irrelevant, as it a horizontally uniform pressure gradient.

Note there is no reference density in Oceananigans because it's dynamically irrelevant and thus has no effect on the momentum balance. You _can_ specify a reference density for a nonlinear equation of state via `SeawaterPolynomials`; in that case the reference density $\rho_0$ determines the magnitude of the buoyancy force, since the nonlinear equation of state predicts $\rho'$ in the expression for buoyancy, $b = - g \rho' / \rho_0$. The reference density can also come into play in thermodynamic calculations (for example air-sea fluxes), but that is left to the user.

Hope that helps! Maybe we can improve the docs...

**Reply by s-aucoin** [reply]
This answers my question, thanks!

**Comment by glwagner** [comment]
Note, the stuff in this discussion and associated docs will be updated by #3080
